File: Python/problem025.py
#! /usr/bin/env python3
"""
Problem 25 - Project Euler
http://projecteuler.net/index.php?section=problems&id=025
"""


def fibonacci():  # generator
    a, b, i = 1, 1, 1
    while True:
        yield i, a
        a, b = b, a + b
        i = i + 1


if __name__ == "__main__":
    maxdigits = 1000
    fib = fibonacci()
    i, f = next(fib)
    while len(str(f)) < maxdigits:
        i, f = next(fib)
    print(i, f)

File: Kotlin/src/main/kotlin/problem025.kt
// Problem 25 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=25
fun fib_digits_index(digits_len: Int): Int {
  var a = 1.toBigInteger()
  var b = 1.toBigInteger()
  var index = 2
  while (b.toString().length < digits_len) {
    val tmp = a
    a = b
    b = b + tmp
    index += 1
  }
  return index
}

fun main() {
  println(fib_digits_index(1000))
}
